### DISCUSSION TITLE
New Profile menu & simplify routes

### DISCUSSION THREAD

**Discussion by EugeneTorap** [apache/superset#21099] [Ideas]
I have a nice idea to rewrite right menu "Settings". I suggest to use a user icon and firstName + lastName. It will be easy and more user friendly to understand which account a user uses.
### Before
![image](https://user-images.githubusercontent.com/29536522/184883494-f42ba422-48ce-4da4-89c4-b28925cb9a88.png)
### After
![image](https://user-images.githubusercontent.com/29536522/184883510-bdfb6cd5-12f4-472d-805b-2b81e28010a1.png)

Second idea, simplify next SPA routes:

- /superset/welcome/ -> /home/
- /superset/profile/ -> /profile/
- /databaseview/list/ -> /database/list/
- /tablemodelview/list/ -> /dataset/list/
- /savedqueryview/list/ -> /saved-query/list/
- /superset/sqllab/history/ -> /query-history/list/
- /superset/sqllab/ -> /sqllab/

**Comment by kgabryje** [comment]
Looks nice!
However, I'm wondering if we shouldn't split user settings from other items from `Settings` menu. Currently we can access user settings, alerts, reports, annotation layers, security settings and more from the `Settings` menu. I think if we used the user icon and name + surname, users might not expect to find anything else than user settings there, which might decrease discoverability of other actions.

As for the URLs, I don't have a strong opinion on that, but the last time we changed route name it turned out to be more "controversial" than we expected. If we were to proceed with changing the route names, we'd need to make sure that the old URLs still work and redirect to the new routes

**Reply by villebro** [reply]
Good point @kgabryje - Currently the Settings menu does indeed have lots of stuff in it beyond the User settings. One could actually argue too much, and it has become a bit of a dumping ground:
- Alerts & Reports
- Annotations
- RLS
- CSS templates
- etc

I wonder if there would be merit in rethinking where these miscellaneous items should be placed..

**Comment by villebro** [comment]
With regard to those routes, the `/superset` prefix applies to legacy routes that will be replaced by new endpoints on the `/api/v1` namespace. So they should not be renamed, but rather refactored to follow the pattern in `/api/v1`.

**Reply by villebro** [reply]
I probably got the SPA routes and API confused here!

**Reply by rusackas** [reply]
It gets a little muddy in there... the /superset URL prefix is indeed used for legacy (non-SPA) views, and I'm all for stripping it out with any view that's moved to SPA. 

Any view that's moved to SPA should also use V1 endpoints, and create new V1 endpoints as needed... following any current URL patterns there as well.

**Comment by villebro** [comment]
@pkdotson @suddjian do you have any preferences on these changes?

**Comment by geido** [comment]
I like the idea of showing the user name. Actually, I am surprised this isn't there but now that you mentioned it I feel like the navbar is incomplete without it. I also agree with the other points above about the discoverability of all the submenu items, but I think that can be solved by separating the settings from the user-related actions. A gear icon just after the user name would do it probably.

**Comment by rusackas** [comment]
I feel like I had a conversation about this years ago, but there was some technical reason we didn't pursue it. I can't remember the reason, so I'm happy to pretend it doesn't exist, and give it another go!

@kasiazjc @jess-dillard any thoughts from your end here?

**Comment by ghost** [comment]
Agree with other folks here that we should separate out user settings from workspace settings if we want to go down this route.
